### PR TITLE
[ADD] pricelist_cache_rest

### PR DESCRIPTION
**Pull request by kv1612**
Provides an endpoint to expose product prices cached by `pricelist_cache`.
Endpoint authorization is based on `auth_api_key`.

### CODE REVIEW

File: pricelist_cache_rest/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Provides an endpoint to get product prices for a given customer
```
Comment (simahawk) [inline pricelist_cache_rest/readme/DESCRIPTION.rst]:
```suggestion
Provides an endpoint to expose product prices cached by `pricelist_cache`.
Endpoint authorization is based on `auth_api_key`.
```
then pls add a `USAGE.rst` module to explain how to use like:

```rst
The endpoint requires the partner ID of the customer prices for which you want to retrieve prices.
It should be called like this::

    curl -X GET "http://your.odoo/pricelist/$partner_id" -H "accept: /" -H "API-KEY: XYZ"

By default you'll get a list of dict like this::

    [
        {"id": $product_id, "price": $price},
        {"id": $product_id, "price": $price},
        {"id": $product_id, "price": $price},
    ]

You can modify the `ir.exports` configuration named "Pricelist Cache Parser" to change this schema.
```
Then add a `CONFIGURATION.rst` to describe how to configure api keys and set them on the company.

**Review by simahawk — CHANGES_REQUESTED** [review]
LG. Docs should be improved.

File: pricelist_cache_rest/wizards/res_config_settings.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+# TODO: this should be moved to `../models`
```
Comment (TDu) [inline pricelist_cache_rest/wizards/res_config_settings.py]: This todo should be attended

**Comment by kv1612** [comment]
Hi @simahawk, @TDu ! Thanks for your review. This PR has been updated.

**Review by mmequignon — APPROVED** [review]

File: pricelist_cache_rest/wizards/res_config_settings.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  <record id="res_config_settings_view_form" model="ir.ui.view">
+    <field name="name">res.config.settings.form.inherit</field>
+    <field name="model">res.config.settings</field>
+    <field name="inherit_id" ref="product.res_config_settings_view_form" />
+    <field name="arch" type="xml">
+      <xpath expr="//field[@name='product_weight_in_lbs']/../../../.." position="after">
+        <div class="col-12 col-lg-6 o_setting_box">
+            <div class="o_setting_left_pane" modifiers="{}" />
+            <div class="o_setting_right_pane" modifiers="{}">
+                <b modifiers="{}">Authored Pricelist Cache API keys</b>
```
Comment (simahawk) [inline pricelist_cache_rest/wizards/res_config_settings.xml]:
```suggestion
                <b modifiers="{}">Authorized Pricelist Cache API keys</b>
```

File: pricelist_cache_rest/readme/CONFIGURATION.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+To configure api keys:
+
+#. Go to `Settings > Units of Measure`.
```
Comment (simahawk) [inline pricelist_cache_rest/readme/CONFIGURATION.rst]: UoM? :thinking: 
We should have a specific box for this IMO.

File: pricelist_cache_rest/readme/CONFIGURATION.rst [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+#. Add your api keys to table `Authored Pricelist Cache API keys`.
```
Comment (simahawk) [inline pricelist_cache_rest/readme/CONFIGURATION.rst]:
```suggestion
#. Add your api keys to table `Authorized Pricelist Cache API keys`.
```

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by kv1612** [comment]
Hi @simahawk, thank for your review. I created another box and put table `Authorized Pricelist Cache API keys` in it. This PR has been updated.

**Review by simahawk — APPROVED** [review]

**Pull request merged**